cdap-docs/admin-manual/source/security/authorization.rst

```diff
@@ -304,90 +303,155 @@ Streams
+    ``dataset_module``.
 
   - For namespaces, ``<entity-id>`` is composed from the namespace, such as
     ``namespace:<namespace-name>``.
 
-  - For artifacts and apps, ``<entity-id>`` is composed of the namespace, entity name, and
-    version, such as ``<namespace-name>.<artifact-name>.<artifact-version>`` or
-    ``<namespace-name>.<app-name>.<app-version>``.
+  - For datasets, streams, artifacts and apps, ``<entity-id>`` is the namespace and entity names, such as
+    ``<namespace-name>.<dataset-name>``, ``<namespace-name>.<stream-name>``, ``<namespace-name>.<artifact-name>``,
+    and ``<namespace-name>.<app-name>``.
 
   - For programs, ``<entity-id>`` includes the application name and the program type:
     ``<namespace-name>.<app-name>.<program-type>.<program-name>``. ``<program-type>`` is
     one of flow, mapreduce, service, spark, worker, or workflow.
 
-  - For datasets and streams, ``<entity-id>`` is the namespace and entity names, such as
-    ``<namespace-name>.<dataset-name>`` or ``<namespace-name>.<stream-name>``.
+  - For datasets, streams, artifacts and apps, ``<entity-id>`` is the namespace and entity names, such as
+    ``<namespace-name>.<dataset-name>``, ``<namespace-name>.<stream-name>``, ``<namespace-name>.<artifact-name>``,
+    and ``<namespace-name>.<app-name>``.
+
+  - ``<principal-type>`` can **only** be ``role`` since Sentry only supports grants privileges to roles.
+
+  - Wildcard can be used in each entity name to grant privileges to multiple entities. For example,
+    ``namespace:ns*`` represents all namespaces that starts with ``ns``.
+    ``namespace:ns?`` represents all namespaces that starts with ``ns`` and follows by a single character.
+    ``program:ns1.app1.*`` represents all types of programs in application app1 in namespace ns1.
+
+- To add the role to other principal, use::
+
+    > add role <role-name> to <principal-type> <principal-name>
+
+  where:
+
+  - ``<role-name>`` is the role name that adds to the principal.
+
+  - ``<principal-type>`` is one of ``role``, ``user`` or ``group``.
+
+- To create a new role, use::
 
-  - For (stream) views, ``<entity-id>`` includes the stream that they were created from:
-    ``<namespace-name>.<stream-name>.<view-name>``.
+    > create role <role-name>
 
 - To check the results, list the privileges for a principal::
 
     > list privileges for <principal-type> <principal-name>
 
+For example,
+to make ``alice`` as the administrator on a namespace ``ns1`` in a new environment, do the following steps:
+- create a new role ``ns1_administrator``
+
+- use the commands to grant *ADMIN* on these entities: ``namespace:ns1``, ``application:ns1.*``, ``program:ns1.*.*``,
+  ``artifact:ns1.*``, ``dataset:ns1.*``, ``stream: ns1.*``, ``dataset_type:ns1.*`` and ``kerberosprincipal.*`` to the role ``ns1_administrator``
+
+- add ``ns1_administrator`` to user ``alice``
```

Comment:
this cannot be done correct ? Since you cannot add user to role in Sentry.

Reply:
I verified in the cli we can do this.